### PR TITLE
Add ETI macros to simplify instantiations for element types and devices for communication helpers

### PR DESCRIPTION
**Pull request by msimberg**
This PR abstracts away the ETI macros a bit more to do all the element types and device types in one macro for communication kernels. Macros for the algorithms will be updated separately.

### CODE REVIEW

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
cscs-ci run

File: include/dlaf/common/eti.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#pragma once
+
+#include <complex>
+
+#include <dlaf/types.h>
+
+#ifdef DLAF_WITH_GPU
+#define DLAF_EXPAND_ETI_SDCZ_DEVICE(ETI_MACRO, KWORD, ...)          \
+  ETI_MACRO(KWORD, double, Device::CPU, __VA_ARGS__);               \
```
Comment (rasolca) [inline include/dlaf/common/eti.h:21]: missing float (it is the source of CI errors for GPU builds...)

**Review by rasolca — COMMENTED** [review]
Looks good

File: include/dlaf/common/eti.h (hunk 1 above)
Comment (msimberg) [reply include/dlaf/common/eti.h:21]: Yep, it was missing for testing and wasn't meant to be pushed...!

**Comment by msimberg** [comment]
cscs-ci run

**Comment by msimberg** [comment]
cscs-ci run

**Review by rasolca — APPROVED** [review]

**Review by albestro — APPROVED** [review]

**Pull request merged**